tests/contrib/futures/test_propagation.py

```diff
@@ -384,3 +437,58 @@ def test_concurrent_futures_with_gevent():
             assert result == 42
         sys.exit(0)
     os.waitpid(pid, 0)
+
+
+def test_submit_no_wait(tracer: DummyTracer):
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=1)
+
+    futures = []
+
+    def work():
+        # This is our expected scenario
+        assert tracer.current_trace_context() is not None
+        assert tracer.current_span() is None
+
+        # DEV: This is the regression case that was raising
+        # tracer.current_span().set_tag("work", "done")
```

Comment:
Should we remove this comment?
```suggestion
        tracer.current_span().set_tag("work", "done")
```

Reply:
no, the comment is here to showcase the example that would cause the exception.

uncommenting will cause these tests to fail.

Reply:
Ahh I misread this test but user's should never use  the result of ` tracer.current_span()` without a None check. `tracer.current_span()` has the type Optional[Span] so `tracer.current_span().set_tag(...)` should log a typing error/warning.